api/v1alpha1/dnsrecord_types.go

```diff
@@ -182,6 +101,37 @@ const (
 	DefaultGeo string = "default"
 )
 
+// GetRootDomain returns the shortest domain that is shared across all spec.Endpoints dns names.
+// Validates that all endpoints share an equal root domain and returns an error if they don't.
+func (s *DNSRecord) GetRootDomain() (string, error) {
+	domain := ""
+	dnsNames := []string{}
+	for idx := range s.Spec.Endpoints {
+		dnsNames = append(dnsNames, s.Spec.Endpoints[idx].DNSName)
+	}
+	for idx := range dnsNames {
+		if domain == "" {
+			domain = dnsNames[0]
+			continue
+		}
+		if len(domain) > len(dnsNames[idx]) {
+			domain = dnsNames[idx]
+		}
+	}
+
+	if domain == "" {
+		return "", fmt.Errorf("unable to determine root domain from %v", dnsNames)
+	}
+
+	for idx := range dnsNames {
+		if !strings.HasSuffix(dnsNames[idx], domain) {
+			return "", fmt.Errorf("inconsitent domains, got %s, expected suffix %s", dnsNames[idx], domain)
+		}
+	}
+
+	return domain, nil
+}
```

Comment:
We might want to replace this by adding a new `rootDomain` field to the dnsrecord spec, but this will work for now.